Imagine/Filter/PostProcessor/CwebpPostProcessor.php

```diff
@@ -0,0 +1,283 @@
+        $this->alphaMethod = $alphaMethod;
+        $this->exact = $exact;
+        $this->metadata = $metadata;
+        $this->resolver = new OptionsResolver();
+
+        $this->configureOptions($this->resolver);
+    }
+
+    public function process(BinaryInterface $binary, array $options = []): BinaryInterface
+    {
+        if (!$this->isBinaryTypeWebpImage($binary)) {
+            return $binary;
+        }
+
+        $file = $this->writeTemporaryFile($binary, $options, 'imagine-post-processor-cwebp');
+        $arguments = $this->getProcessArguments($options);
+        $arguments[] = $file;
+        $arguments[] = '-o';
+        $arguments[] = '-';
+        $process = $this->createProcess($arguments, $options);
+
+        $process->run();
+
+        if (!$this->isSuccessfulProcess($process)) {
+            unlink($file);
+
+            throw new ProcessFailedException($process);
+        }
+
+        $result = new Binary($process->getOutput(), $binary->getMimeType(), $binary->getFormat());
+
+        unlink($file);
+
+        return $result;
+    }
+
+    protected function isBinaryTypeWebpImage(BinaryInterface $binary): bool
+    {
+        return $this->isBinaryTypeMatch($binary, ['image/webp']);
+    }
+
+    protected function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver
+            ->define('q')
+            ->default($this->q)
+            ->allowedTypes('null', 'int')
+            ->allowedValues(static function ($value) {
+                if (null === $value) {
+                    return true;
+                }
+
+                return $value >= 0 && $value <= 100;
+            })
+        ;
+
+        $resolver
+            ->define('alphaQ')
+            ->default($this->alphaQ)
+            ->allowedTypes('null', 'int')
+            ->allowedValues(static function ($value) {
+                if (null === $value) {
+                    return true;
+                }
+
+                return $value >= 0 && $value <= 100;
+            })
+        ;
+
+        $resolver
+            ->define('m')
+            ->default($this->m)
+            ->allowedTypes('null', 'int')
+            ->allowedValues(static function ($value) {
+                if (null === $value) {
+                    return true;
+                }
+
+                return $value >= 0 && $value <= 6;
+            })
+        ;
+
+        $resolver
+            ->define('alphaFilter')
+            ->default($this->alphaFilter)
+            ->allowedTypes('null', 'string')
+            ->allowedValues(static function ($value) {
+                if (null === $value) {
+                    return true;
+                }
+
+                return \in_array($value, ['none', 'fast', 'best'], true);
+            })
+        ;
+
+        $resolver
+            ->define('alphaMethod')
+            ->default($this->alphaMethod)
+            ->allowedTypes('null', 'int')
+            ->allowedValues(static function ($value) {
```

Comment:
could you provide `[null, 0, 1]` instead of a callback? similar above for alphaFilter? (m and the ones with values from 1-100 are fine with the callback. and metadata is a more complex structure, so lets keep that as is too)

Reply:
Done. :)